testsuite/tests/prometheus/apicast/selfmanaged/conftest.py

```diff
@@ -0,0 +1,46 @@
+"""Provide custom gateway for tests changing apicast parameters."""
+
+from weakget import weakget
+import pytest
+
+from testsuite.gateways import gateway
+from testsuite.gateways.apicast.template import TemplateApicast
+from testsuite.utils import blame, warn_and_skip
+
+
+@pytest.fixture(scope="module", autouse=True)
+def require_openshift(testconfig):
+    """These tests require openshift available"""
+    if not weakget(testconfig)["openshift"]["servers"]["default"] % False:
+        warn_and_skip("All monitoring for selfmanaged apicast skipped due to missing openshift")
+
+
+@pytest.fixture(scope="module")
+def staging_gateway(request, gateway_kind, gateway_environment, gateway_options):
+    """Deploy self-managed template based apicast gateway."""
+    gw = gateway(kind=gateway_kind, staging=True, name=blame(request, "gw"), **gateway_options)
+    request.addfinalizer(gw.destroy)
+    gw.create()
+
+    if len(gateway_environment) > 0:
```

Comment:
If you don't need an environment in subsequent tests you don't need to use it.

Reply:
I removed thinks that I do not use, please check if my cleanup makes sense.